src/frontend/i18n_phase.nit

```diff
@@ -131,6 +131,39 @@ redef class AStringExpr
 	end
 end
 
+redef class ASuperstringExpr
+
+	redef fun accept_string_finder(v) do
+		var fmt = ""
+		var exprs = new Array[AExpr]
+		for i in n_exprs do
+			if i isa AStringFormExpr then
+				fmt += i.value.as(not null)
+			else
+				fmt += "%"
+				exprs.push i
+				fmt += exprs.length.to_s
+			end
+		end
+		var loc = location
+		var locstr = "{v.amodule.mmodule.mgroup.name}::{v.amodule.mmodule.name} {loc.line_start}--{loc.column_start}:{loc.column_end}"
+		if not v.strings.has_key(fmt) then
+			var po = new PObject([locstr], fmt, "")
+			v.strings[fmt] = po
+		else
+			v.strings[fmt].locations.push locstr
+		end
```

Comment:
instead of copying from AStringExpr I propose that you move most if the previous lines to the visitor `v`